src/linux/agent/system-server.js

```diff
@@ -85,12 +86,35 @@ rpc.exports = {
   },
   startActivity: function (pkg, activity) {
     return performOnJavaVM(function () {
-      var intent = packageManager.getLaunchIntentForPackage(pkg);
-      if (intent === null)
+      var appInstalled = false;
+      var apps = packageManager.getInstalledApplications(GET_META_DATA);
+      for (var i = 0; i < apps.size(); i++) {
+        var appInfo = Java.cast(apps.get(i), ApplicationInfo);
+        if (appInfo.packageName.value === pkg) {
+          appInstalled = true;
+          break;
+        }
+      }
+      if (!appInstalled)
         throw new Error("Unable to find application with identifier '" + pkg + "'");
 
-      if (activity !== null)
+      var intent = packageManager.getLaunchIntentForPackage(pkg);
+      if (intent === null && activity === null)
+        throw new Error("Unable to find a front-door activity");
```

Comment:
Strings should be this style:

```suggestion
        throw new Error('Unable to find a front-door activity');
```

(Though we've made a few exceptions in this file for cases where we want to use single-quotes in the string literal.)